database.go

```diff
@@ -312,8 +312,8 @@ func (m *Measurement) idsForExpr(n *influxql.BinaryExpr) (seriesIDs, bool, influ
 // value should be included in the resulting set, and an expression if the return is a field expression.
 // The map that it takes maps each series id to the field expression that should be used to evaluate it when iterating over its cursor.
 // Series that have no field expressions won't be in the map
-func (m *Measurement) walkWhereForSeriesIds(node influxql.Node, filters map[uint32]influxql.Expr) (seriesIDs, bool, influxql.Expr) {
-	switch n := node.(type) {
+func (m *Measurement) walkWhereForSeriesIds(expr influxql.Expr, filters map[uint32]influxql.Expr) (seriesIDs, bool, influxql.Expr) {
```

Comment:
This function is `walkWhere...` meaning "walk the where clause".  A where clause is of type `influxql.Expr` and it's not valid to pass in any other AST node type.  This change makes that clearer and allows better compile time type checking.